modules/dnn/include/opencv2/dnn/layer.hpp

```diff
@@ -66,6 +66,9 @@ class CV_EXPORTS LayerFactory
     //! Unregisters registered layer with specified type name. Thread-safe.
     static void unregisterLayer(const String &type);
 
+    //! Get the types of layers which were registered.
+    static const std::map<String, int>& getRegisteredLayers();
```

Comment:
> `std::map<String, int>`

Why do we need `std::map` in public API? Use `std::vector` instead.

> String

Use `std::string` in new API.

---

If you need to optimize access with `O(log(N))` instead `O(N)`, then build `std::set` / hash_map / etc internally. Do not expose that to public API.

---

`std::set` does not work with OpenCV bindings, so it should be avoided in public API.

Reply:
Corrected, added [layer_reg.private.hpp](https://github.com/opencv/opencv/pull/19693/files#diff-72def3133939a67ff89fd1fe6b97e94f4bb697472f71aa7de45d822998b4499c).